packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -110,13 +110,28 @@ bool ArbCore::machineIdle() {
 }
 
 ArbCore::message_status_enum ArbCore::messagesStatus() {
+    auto lock = std::shared_lock<std::shared_mutex>(message_data_error_mutex);
     auto current_status = message_data_status.load();
-    if (!core_error && current_status != MESSAGES_READY) {
+    if (current_status != MESSAGES_ERROR && current_status != MESSAGES_READY) {
         message_data_status = MESSAGES_EMPTY;
     }
+
     return current_status;
 }
 
+std::string ArbCore::messagesClearError() {
+    auto lock = std::shared_lock<std::shared_mutex>(message_data_error_mutex);
```

Comment:
Doesn't this need to be a unique lock as it mutates the `message_data_error_string`? If so, and we remove the lock in `messagesStatus`, we should probably make this a regular non-shared mutex.

Reply:
as discussed, mutex can be removed